### ISSUE TITLE
Cloudflare proxy support

### ISSUE THREAD

**Issue by 23doors** [zitadel/terraform-provider-zitadel#159]
### Preflight Checklist

- [X] I could not find a solution in the documentation, the existing issues or discussions
- [X] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Version

1.0.5

### Describe the problem caused by this bug

When using proxied subdomain for zitadel in cloudflare I can't seem to use terraform zitadel provider. Possibly also zitadel sdk in general as I guess it uses that.

Enabled grpc support on cloudflare side but it requires content-type to be application/grpc. Zitadel seems to be returning an incorrect content-type. Getting:
`520 (); transport: received unexpected content-type "text/plain; charset=UTF-8"`

### To reproduce

Pretty self explanatory.

1. Use zitadel with cloudflare proxied domain.
1. Enable grpc on cloudflare side.
1. Try to do anything in zitadel terraform provider.

### Expected behavior

Cloudflare supports proxying grpc so it should work normally. It seems to be a matter of wrong content-type being used.

### Additional Context

TF_LOG=trace logs

```
2024-02-19T13:40:44.455+0100 [ERROR] provider.terraform-provider-zitadel_v1.0.5: Response contains error diagnostic: @caller=github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov6/internal/diag/diagnostics.go:55 diagnostic_summary="error while getting org by id ICS: rpc error: code = Unknown desc = unexpected HTTP status code received from server: 520 (); transport: received unexpected content-type \"text/plain; charset=UTF-8\"" tf_provider_addr=registry.terraform.io/zitadel/zitadel tf_proto_version=6.3 diagnostic_detail="" tf_data_source_type=zitadel_orgs tf_req_id=934c9b50-b738-673c-1804-6fb77844f482 @module=sdk.proto diagnostic_severity=ERROR tf_rpc=ReadDataSource timestamp="2024-02-19T13:40:44.454+0100"
2024-02-19T13:40:44.455+0100 [TRACE] provider.terraform-provider-zitadel_v1.0.5: Served request: tf_data_source_type=zitadel_orgs tf_provider_addr=registry.terraform.io/zitadel/zitadel tf_req_id=934c9b50-b738-673c-1804-6fb77844f482 @module=sdk.proto tf_proto_version=6.3 tf_rpc=ReadDataSource @caller=github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov6/tf6server/server.go:668 timestamp="2024-02-19T13:40:44.454+0100"
2024-02-19T13:40:44.455+0100 [ERROR] vertex "data.zitadel_orgs.zitadel_ics_org" error: error while getting org by id ICS: rpc error: code = Unknown desc = unexpected HTTP status code received from server: 520 (); transport: received unexpected content-type "text/plain; charset=UTF-8"
2024-02-19T13:40:44.455+0100 [TRACE] vertex "data.zitadel_orgs.zitadel_ics_org": visit complete, with errors
2024-02-19T13:40:44.455+0100 [TRACE] dag/walk: upstream of "root" errored, so skipping
2024-02-19T13:40:44.455+0100 [TRACE] vertex "data.zitadel_orgs.zitadel_ics_org (expand)": dynamic subgraph encountered errors: error while getting org by id ICS: rpc error: code = Unknown desc = unexpected HTTP status code received from server: 520 (); transport: received unexpected content-type "text/plain; charset=UTF-8"
```

When disabled cloudflare proxying, it works fine.

**Comment by vavsab** [comment]
I had a similar issue with cloudflare. Maybe my case helps you.

TL DR;
Pressing "Enable GRPC" checkbox in cloudflare is not enough 😭. You need to struggle more to make it work.

I'm using cloudflare tunnels. When I tried to setup a sample GRPC server I could not query it through cloudflare.
So I spent a couple of weeks playing with this https://github.com/cloudflare/cloudflared/issues/491
And after my sample GRPC server was working - terraform also started to work.

**Comment by 23doors** [comment]
Not sure if this relates to this issue to be honest. I'm not using cloudflare tunnels, they're completely different from normal proxying. And briefly checking issue you linked, it doesn't exactly provide any solution to try either. 

According to docs, Enable GRPC checkbox for cloudflare proxied resources (not argo tunnels) is actually enough.
